src/main.tf

```diff
@@ -118,6 +118,57 @@ resource "azurerm_linux_function_app" "starter" {
   }
 }
 
+locals {
+  default_frontend_endpoint_name      = "${var.naming_prefix}-${var.environment}-fd-azurefd-net"
+  default_frontend_endpoint           = "${var.naming_prefix}-${var.environment}-fd.azurefd.net"
+}
+
+resource "azurerm_frontdoor" "starter" {
+  name                = "${var.naming_prefix}-${var.environment}-fd"
+  friendly_name       = "${var.naming_prefix}-${var.environment}-fd"
+  resource_group_name = data.azurerm_resource_group.starter.name
+
+  frontend_endpoint {
+    name      = local.default_frontend_endpoint_name
+    host_name = local.default_frontend_endpoint
+  }
+
```

Comment:
Another idea you could implement to have both the front end and backends behind Azure Front Door would be to use a path-based routing rule (rather than multiple frontend domains). 

For e.g.: You could route all requests for path 'api/*' to the Azure Functions backend and all other requests to the Azure Web Apps backend. Then you could use the same domain for both.